homeassistant/components/deako/__init__.py

```diff
@@ -0,0 +1,62 @@
+"""The deako integration."""
+
+from __future__ import annotations
+
+import logging
+
+from pydeako.deako import Deako, DeviceListTimeout, FindDevicesTimeout
+from pydeako.discover import DeakoDiscoverer
+
+from homeassistant.components import zeroconf
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import DOMAIN
+
+_LOGGER: logging.Logger = logging.getLogger(__name__)
+
+PLATFORMS: list[Platform] = [Platform.LIGHT]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up deako."""
+    _zc = await zeroconf.async_get_instance(hass)
+    discoverer = DeakoDiscoverer(_zc)
+
+    connection = Deako(discoverer.get_address)
+
+    await connection.connect()
+    try:
+        await connection.find_devices()
+    except DeviceListTimeout as exc:  # device list never received
+        _LOGGER.warning("Device not responding to device list")
+        await connection.disconnect()
+        raise ConfigEntryNotReady(exc) from exc
+    except FindDevicesTimeout as exc:  # total devices expected not received
+        _LOGGER.warning("Device not responding to device requests")
+        await connection.disconnect()
+        raise ConfigEntryNotReady(exc) from exc
+
+    # If deako devices are advertising on mdns, we should be able to get at least one device
+    devices = connection.get_devices()
+    if len(devices) == 0:
+        await connection.disconnect()
+        raise ConfigEntryNotReady(devices)
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = connection
```

Comment:
Let's use the `entry.runtime_data`, there's a dev blog post about that a month or 2 ago, I'm on mobile so I dont have a link

Reply:
Ooooo I like that. Thanks! Certainly cleans things up a bit.